### PR TITLE
67 leaderboard

### LINKED ISSUE #67: LeaderBoard in Hero Section

### PR DESCRIPTION
**Pull request by DLittlefield81**
mock data is imported, sorted by totalbadges and then sliced to only have the highest badge totals

### CODE REVIEW

**Review by JWMedeiros — DISMISSED** [review]
This also looks good to me, but keep in mind there might be merge conflicts in Hero.tsx and Leaderboard.tsx depending on who merges first as Daniel is re-doing import paths.

File: src/components/Hero/Hero.tsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 'use client'
 
 import styles from './Hero.module.css'
+import Leaderboard from '../Leaderboard/Leaderboard'
```
Comment (rmc3408) [inline src/components/Hero/Hero.tsx]: use Leaderboard/index.tsx and export/import. Look other part of the code

File: src/components/Leaderboard/Leaderboard.module.css [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
   border: 4px solid #fafafa;
   background: rgba(250, 250, 250, 0.4);
   width: 340px;
 }
 
+.alignRight {
+  float: right;
+}
+
 .h2 {
```
Comment (rmc3408) [inline src/components/Leaderboard/Leaderboard.module.css]: better naming... maybe Title.

File: src/components/Leaderboard/Leaderboard.tsx [hunk 3]
```diff
@@ -4,83 +4,31 @@ import Table from '@mui/material/Table'
...
-  const handleChangePage = (event: unknown, newPage: number) => {
-    setPage(newPage)
-  }
-
-  // const handleChangeRowsPerPage = (event: React.ChangeEvent<HTMLInputElement>) => {
-  //     setRowsPerPage(+event.target.value);
-  //     setPage(0);
-  // };
-
+// Set Leaderboard Record Limit
```
Comment (rmc3408) [inline src/components/Leaderboard/Leaderboard.tsx]: no comments

File: src/components/Leaderboard/Leaderboard.tsx [hunk 4]
```diff
@@ -4,83 +4,31 @@ import Table from '@mui/material/Table'
...
+const sortedHighScores = [...userData].sort((a, b) => b.totalBadge - a.totalBadge)
+const highScores = sortedHighScores.slice(0, 5)
```
Comment (rmc3408) [inline src/components/Leaderboard/Leaderboard.tsx]: no magic numbers... number 5. What happens if you dont have 5 ?
missing type of highscores, 
missing types in line 12.

File: src/components/Hero/Hero.tsx [hunk 5]
```diff
@@ -15,7 +16,9 @@ const Hero = () => {
           <p className={styles.text}>{HERO_DESCRIPTION}</p>
         </div>
       </div>
-      <div className={styles.col}>{/* Leaderboard Component */}</div>
+      <div className={styles.col}>
```
Comment (rmc3408) [inline src/components/Hero/Hero.tsx]: on line 10,  call like leaderboard, not arrow function

File: src/components/Leaderboard/Leaderboard.tsx [hunk 6]
```diff
@@ -4,83 +4,31 @@ import Table from '@mui/material/Table'
...
-                </TableRow>
-              )
-            })}
-          </TableBody> */}
+          <TableBody>
+            {highScores.map((user) => (
+              <TableRow key={user.id}>
+                <TableCell className={styles.tableText}>
+                  <h3>
+                    {user.username}
```
Comment (rmc3408) [inline src/components/Leaderboard/Leaderboard.tsx]: use object destructuring for user.username and totalbadges

**Review by rmc3408 — COMMENTED** [review]
cleaning and add types
Please return the CICD that was deleted. `.github/workflows`

File: src/components/Leaderboard/Leaderboard.module.css (hunk 2 above)
Comment (DLittlefield81) [reply src/components/Leaderboard/Leaderboard.module.css]: changed: badgeAlignRight

File: src/components/Leaderboard/Leaderboard.tsx (hunk 4 above)
Comment (DLittlefield81) [reply src/components/Leaderboard/Leaderboard.tsx]: May need assist. Look at what I've changed, if it's not right, I need guidance.

File: src/components/Hero/Hero.tsx (hunk 1 above)
Comment (DLittlefield81) [reply src/components/Hero/Hero.tsx]: added index file

**Review by rmc3408 — APPROVED** [review]
LGTM

**Pull request merged**